### PR TITLE
Remove providers with no vehicles after X minutes since last update

### PR DESCRIPTION
**Pull request by vessla**

### CODE REVIEW

File: src/main/java/org/opentripplanner/routing/graph/Graph.java [hunk 1]
```diff
@@ -527,6 +533,17 @@ public void addTransitRoutes(Collection<Route> routes) {
         return transitRoutes;
     }
 
+    public Map<Provider, LocalTime> getLastProviderVehiclesUpdateTimestamps() {
+        return lastProviderVehiclesUpdateTimestamps;
+    }
+
+    public boolean isUnresponsiveGracePeriodExceeded(Provider provider, LocalTime currentUpdateTimestamp) {
```
Comment (MironSz) [inline src/main/java/org/opentripplanner/routing/graph/Graph.java:540]: I'm not sure if passing 'currentUpdateTimestamp' is necessary

File: src/main/java/org/opentripplanner/hasura_client/mappers/VehiclePositionsMapper.java [hunk 2]
```diff
@@ -7,12 +7,16 @@
...
 import java.math.BigDecimal;
+import java.util.HashMap;
+import java.util.Map;
 import java.util.Optional;
 
 public class VehiclePositionsMapper extends HasuraToOTPMapper<Vehicle, VehicleDescription> {
 
     private static final Logger LOG = LoggerFactory.getLogger(HasuraGetter.class);
 
+    private Map<Provider, Long> numberOfMappedVehiclesPerProvider = new HashMap<>();
```
Comment (MironSz) [inline src/main/java/org/opentripplanner/hasura_client/mappers/VehiclePositionsMapper.java]: Wouldn't Set<Provider> be more straight forward? Mapping from providers to number of fetched vehicles isn't needed anywhere.

Comment (vessla) [reply src/main/java/org/opentripplanner/hasura_client/mappers/VehiclePositionsMapper.java]: Good idea - done. My initial plan was to perhaps also take into account, how many of the vehicles have disappeared, however at this point the exact number of vehicles per provider is indeed not required.

File: src/main/java/org/opentripplanner/routing/graph/Graph.java (hunk 1 above)
Comment (vessla) [reply src/main/java/org/opentripplanner/routing/graph/Graph.java:540]: The idea behind it was to make sure, that I evaluated all the vehicles from providers using the same criteria, however keeping in mind, that the entire update process takes ~1.8 second, it probably does not make much difference...

**Review by MironSz — APPROVED** [review]

**Pull request merged**